cura/Settings/ContainerManager.py

```diff
@@ -29,21 +29,115 @@ def __init__(self, parent = None):
     #   \return The ID of the new container, or an empty string if duplication failed.
     @pyqtSlot(str, result = str)
     def duplicateContainer(self, container_id):
-        containers = self._registry.findInstanceContainers(id = container_id)
+        containers = self._registry.findContainers(None, id = container_id)
         if not containers:
-            return
+            return ""
+
+        container = containers[0]
+
+        new_container = None
+        new_name = self._registry.uniqueName(container.getName())
+        if hasattr(container, "duplicate"):
```

Comment:
It is a function that only exists for InstanceContainer at the moment. I will add a bit of explanation.